### PR TITLE
Adding procs inventory

### PR DESCRIPTION
**Pull request by CallisteH**
Work in progress 

We add virtual processors and entitled processors in the LPAR inventory.

We also plan to implement whether the VM is dedicated or shared.

### CODE REVIEW

File: app/models/manageiq/providers/ibm_power_hmc/inventory/parser/infra_manager.rb [hunk 1]
```diff
@@ -207,6 +207,37 @@ def parse_vm_advanced_settings(vm, lpar)
...
+        :name         => 'entitled_processors',
+        :display_name => _('Entitled Processors'),
+        :description  => _('The number of entitled processors assigned to the VM'),
+        :value        => lpar.proc_units,
+        :read_only    => true
+      )
+    end
+
+    $ibm_power_hmc_log.info("#{self.class}##{__method__} : lpar.dedicated = #{lpar.dedicated}")
+    if lpar.dedicated.eql? "true"
```
Comment (jaywcarman) [inline app/models/manageiq/providers/ibm_power_hmc/inventory/parser/infra_manager.rb:222]: You could do this without duplicating all of the common values:

```suggestion
    proc_type = lpar.dedicated ==  "true" ? "dedicated" : lpar.sharing_mode
    persister.vms_and_templates_advanced_settings.build(
      :resource     =>vm,
      :name         => 'processor_type',
      :display_name => _('Processor type'),
      :description  => _('dedicated: Dedicated, shared: Uncapped shared, capped: Capped shared'),
      :value        => proc_type,
      :read_only    => true
    )
```

File: app/models/manageiq/providers/ibm_power_hmc/inventory/parser/infra_manager.rb [hunk 2]
```diff
@@ -207,6 +207,37 @@ def parse_vm_advanced_settings(vm, lpar)
...
+    else
+      persister.vms_and_templates_advanced_settings.build(
+        :resource     => vm,
+        :name         => 'processor_type',
+        :display_name => _('Processor type'),
+        :description  => _('dedicated: Dedicated, shared: Uncapped shared, capped: Capped shared'),
+        :value        => lpar.sharing_mode,
+        :read_only    => true
+      )
+    end
```
Comment (jaywcarman) [inline app/models/manageiq/providers/ibm_power_hmc/inventory/parser/infra_manager.rb:240]: You could do this without duplicating all of the common values:

```suggestion
    proc_type = lpar.dedicated ==  "true" ? "dedicated" : lpar.sharing_mode
    persister.vms_and_templates_advanced_settings.build(
      :resource     =>vm,
      :name         => 'processor_type',
      :display_name => _('Processor type'),
      :description  => _('dedicated: Dedicated, shared: Uncapped shared, capped: Capped shared'),
      :value        => proc_type,
      :read_only    => true
    )
```

**Comment by jaywcarman** [comment]
> Waiting for #38

fyi #38 is now merged.

**Comment by CallisteH** [comment]
If it's good for you @jaywcarman , I think we can merge

**Review by jaywcarman — APPROVED** [review]

**Comment by jaywcarman** [comment]
Checking in to see if this has made it through the `najdorf/yes?` review process.

**Comment by Fryguy** [comment]
@CallisteH A conflict occurred during the backport of this pull request to `najdorf`.

If this pull request is based on another pull request that has not been marked for backport, add the appropriate labels to the other pull request. Otherwise, please create a new pull request direct to the `najdorf` branch in order to resolve this.

Conflict details:

```diff
diff --cc manageiq-providers-ibm_power_hmc.gemspec
index 77bed7f,fa4c492..0000000
--- a/manageiq-providers-ibm_power_hmc.gemspec
+++ b/manageiq-providers-ibm_power_hmc.gemspec
@@@ -19,7 -19,7 +19,11 @@@ Gem::Specification.new do |spec
    spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
    spec.require_paths = ["lib"]
  
++<<<<<<< HEAD
 +  spec.add_dependency "ibm_power_hmc", "~> 0.8.4"
++=======
+   spec.add_dependency "ibm_power_hmc", "~> 0.8.3"
++>>>>>>> 3f07cbe (Merge pull request #37 from CallisteH/feature/lpar_procs)
  
    spec.add_development_dependency "manageiq-style"
    spec.add_development_dependency "simplecov", ">= 0.21.2"
```

**Comment by Fryguy** [comment]
I will backport this manually.

**Comment by jaywcarman** [comment]
The conflict is caused since we backported #39 first, which requires `"ibm_power_hmc", "~> 0.8.4"`.

**Comment by Fryguy** [comment]
`najdorf` backport details:

```
commit c98738b07602b4f0c2acd1f38a01bf1e04170997 (HEAD -> najdorf, upstream/najdorf)
Author: Jay Carman <jwcarman@us.ibm.com>
Date:   Tue Feb 15 13:07:00 2022 -0600

    Merge pull request #37 from CallisteH/feature/lpar_procs

    Adding procs inventory

    (cherry picked from commit 3f07cbe3b969fb5b393262226bf8ad8247cb40ae)

    Conflicts:
      manageiq-providers-ibm_power_hmc.gemspec:
        Removed the gemspec reference because najdorf was already at a  newer version
```

**Pull request merged**